### PR TITLE
Add TypeSpec-to-SDK workflow skills

### PR DESCRIPTION
**Pull request by ronniegeraghty**
## Summary

Adds 4 waza-based Copilot skills for the TypeSpec-to-SDK release workflow.

### Skills Added (.github/skills/)

- **sdk-release** — Check SDK package release readiness and trigger the release pipeline. Shows pipeline link and informs user about release stage approval.
- **prepare-release-plan** — Create and manage release plan work items for Azure SDK releases across languages.
- **pipeline-troubleshooting** — Diagnose and resolve failures in Azure SDK CI pipelines and SDK generation pipelines.
- **apiview-feedback-resolution** — Retrieve and resolve feedback from APIView reviews on Azure SDK packages.

Each skill includes:
- `SKILL.md` with waza-compliant frontmatter (name, license, description with routing markers)
- `eval.yaml` with task-based evaluation suite
- `references/` for supporting documentation
- `fixtures/` for test data (where applicable)
- `tasks/` with eval scenarios (basic usage, edge cases, negative tests)

### What was removed per review feedback

13 skills were removed from this PR based on reviewer feedback:
- **8 deleted**: validate-service-label, verify-setup, validate-codeowners, typespec-to-sdk-workflow, typespec-customization, typespec-authoring, package-release-readiness, check-package-validation
- **2 deferred to azure-sdk-tools repo**: generate-sdk-locally, report-test-bug (will be shared across repos with `shared-` prefix)
- **3 deferred to separate PR**: sensei, skill-authoring, markdown-token-optimizer (skill development tools)

Also removed from this PR:
- `skills-eval.yml` workflow (separate PR)
- `.waza.yaml` config
- `mcp.json` and `cspell.yaml` changes (reverted to match main)

### CODE REVIEW

File: .github/skills/file-test-bug/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+name: file-test-bug
```
Comment (praveenkuttappan) [inline .github/skills/file-test-bug/SKILL.md]:
```suggestion
name: report-test-bug
```

**Review by ronniegeraghty — COMMENTED** [review]
Done — renamed `file-test-bug` to `report-test-bug` (directory, SKILL.md name field, and README references) in commit c037bbe.

File: .github/skills/file-test-bug/SKILL.md (hunk 1 above)
Comment (ronniegeraghty) [reply .github/skills/file-test-bug/SKILL.md]: Done — renamed file-test-bug to report-test-bug (directory, SKILL.md name field, and README references) in commit https://github.com/Azure/azure-rest-api-specs/commit/c037bbe1446e074db6ee077329a91bbcac2723bb.

File: .github/skills/validate-service-label/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/validate-service-label/SKILL.md:1]: Let's remove this skill.  These were the instructions added by interns for service code owner tools. Daniel is working on a different approach for labels so we should completely ignore label related skills for now.

File: .github/skills/verify-setup/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/verify-setup/SKILL.md:1]: Does this need to be a skill? verify setup runs only one mcp tool call and it can be direct reeference in AGENTS.md.  @JennyPng has done some evaluation earlier to try skill for verify setup and found no value addition to create a skill for verify setup. I would prefer to have an instruction in AGENTS.md to run verify setup. Or it can be plugged into local SDK generation skill.

File: .vscode/mcp.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
       "azure-sdk-mcp": {
         "type": "stdio",
         "command": "pwsh",        
-        "args": ["./eng/common/mcp/azure-sdk-mcp.ps1", "-Run"]
-    }
+        "args": ["${workspaceFolder}/eng/common/mcp/azure-sdk-mcp.ps1", "-Run"]
```
Comment (praveenkuttappan) [inline .vscode/mcp.json]: Can you please revert this change? copilot-cli cannot process this mcp.json.

File: cspell.yaml [hunk 3]
```diff
@@ -59,6 +59,24 @@ words:
...
+      - evals
+      - frontmatter
+      - mygroup
+      - sensei
+      - spboyer
+      - waza
+  - filename: '.github/workflows/skills-eval.yml'
+    words:
+      - waza
+      - spboyer
```
Comment (praveenkuttappan) [inline cspell.yaml:68]: alias? Why do we even have alias in yml?

Comment (praveenkuttappan) [reply cspell.yaml:68]: I see. You are using the published registry. Can we check with him to publish it to a more org level github account so we don't have to pull it from his GitHub account? If possible, let's split this PR into two PRs.
1. Just for the skils
2. Workflow to evaluate the skills

File: .github/skills/validate-codeowners/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/validate-codeowners/SKILL.md:1]: Let's remove this skill for now and wait until Daniel completes code owner project.

File: .github/skills/typespec-to-sdk-workflow/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/typespec-to-sdk-workflow/SKILL.md:1]: I think we should put this high-level workflow into AGENTS.md instead of creating as a skill. This makes it to have a high chance of always referring the workflow seps.

Comment (praveenkuttappan) [reply .github/skills/typespec-to-sdk-workflow/SKILL.md:1]: GitHub agent workflow can take care of converting an experimenting release plan to actual release plan so we don't need to have a skill to transition a release plan.

File: .github/skills/typespec-customization/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/typespec-customization/SKILL.md:1]: Let's remove this one from this PR. Sameeksha has a PR for TypeSpec customization. We can make additional fine tuning in that skill to support front matters and required files for eval.

File: .github/skills/typespec-authoring/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/typespec-authoring/SKILL.md:1]: Let's remove TypeSpec authoring skill from this PR. Haoling is working on it.

File: .github/skills/sensei/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+- Adding trigger phrases and anti-triggers to skill descriptions
+- Batch-improving multiple skills at once
+- Auditing and fixing Low-adherence skills
+
+## Invocation Modes
+
+### Single Skill
+
+```
+Run sensei on azure-deploy
```
Comment (praveenkuttappan) [inline .github/skills/sensei/SKILL.md]: This looks like still has reference to GH copilot for Azure.

File: .github/skills/sensei/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+║                                                                  ║
+║  USAGE:                                                          ║
+║    Run sensei on <skill-name>              # Single skill        ║
+║    Run sensei on <skill-name> --skip-integration  # Fast mode    ║
+║    Run sensei on <skill1>, <skill2>, ...   # Multiple skills     ║
+║    Run sensei on all Low-adherence skills  # Batch by score      ║
+║    Run sensei on all skills                # All skills       ║
+║                                                                  ║
+║  EXAMPLES:                                                       ║
+║    Run sensei on appinsights-instrumentation                     ║
```
Comment (praveenkuttappan) [inline .github/skills/sensei/SKILL.md]: This skill requires changes to refer more about Azure SDK tools and not azure functionality. One possibility is to move skills eval and sensei testing to a separate PR.

File: .github/skills/sdk-release/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+## Steps
+
+1. **Collect Info** — Get `packageName` and `language` from the user. Optionally get the `branch` (defaults to main).
+2. **Check Readiness** — Run `azsdk_release_sdk` with `checkReady: true` to verify:
+   - API review approval status
+   - Changelog completeness
+   - Package name approval (for new preview packages)
+   - Release date set in release tracker
+3. **Review Results** — If not ready, display failing checks and guide the user to resolve issues.
+4. **Trigger Release** — Once ready, run `azsdk_release_sdk` with `checkReady: false` to trigger the pipeline. Inform user they must approve the release stage after triggering.
```
Comment (praveenkuttappan) [inline .github/skills/sdk-release/SKILL.md]:
```suggestion
4. **Trigger Release** — Once ready, run `azsdk_release_sdk` with `checkReady: false` to trigger the pipeline. Show the release pipeline link to user and inform user they must approve the release stage after triggering. 
```

File: .github/skills/report-test-bug/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
+---
+name: report-test-bug
+description: "File a GitHub issue for local integration test failures. TRIGGERS: file test bug, report test failure, create bug for test, integration test failed, test failure issue, junit failure"
+---
+
+# File Test Bug
+
+Creates a GitHub issue in `microsoft/github-copilot-for-azure` for integration test failures.
```
Comment (praveenkuttappan) [inline .github/skills/report-test-bug/SKILL.md]: Wrong repo name.

File: .github/skills/report-test-bug/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/report-test-bug/SKILL.md:1]: We need to rename these skill folders to have prefix `shared-` and this should be in azure-sdk-tools repo so it can be synced across all the repos.

File: .github/skills/report-test-bug/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+## Input
+
+- **Skill name** (required): e.g., `azure-rbac`, `appinsights-instrumentation`
+- **Test run** (optional): Timestamp of test run. Defaults to most recent with logs for the skill.
+
+## Steps
+
+1. Ask user for skill name if not provided
```
Comment (praveenkuttappan) [inline .github/skills/report-test-bug/SKILL.md]: Why is skill name is required as input?

File: .github/skills/report-test-bug/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+description: "File a GitHub issue for local integration test failures. TRIGGERS: file test bug, report test failure, create bug for test, integration test failed, test failure issue, junit failure"
+---
+
+# File Test Bug
+
+Creates a GitHub issue in `microsoft/github-copilot-for-azure` for integration test failures.
+
+## Input
+
+- **Skill name** (required): e.g., `azure-rbac`, `appinsights-instrumentation`
```
Comment (praveenkuttappan) [inline .github/skills/report-test-bug/SKILL.md]: Azure specific terms

File: .github/skills/report-test-bug/SKILL.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+   - Compare agent behavior to what the test expected (from the assertion)
+   - Identify the root cause (skill issue, test issue, or model behavior)
+   - Write 2-3 sentences per failed test explaining WHY it failed
+   - Suggest potential fixes (update skill, update test, or update fixtures)
+7. Create issue via `github-mcp-server-create_issue`:
+
+````
+owner: microsoft
+repo: github-copilot-for-azure
+title: Integration test failure in <skill-name>
```
Comment (praveenkuttappan) [inline .github/skills/report-test-bug/SKILL.md]: Wrong repo name for Azure sdk tools

File: .github/skills/package-release-readiness/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/package-release-readiness/SKILL.md:1]: Let's remove this skill. This is no longer required and it's part of package release skill.

File: .github/skills/markdown-token-optimizer/SKILL.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+---
+name: markdown-token-optimizer
+description: "Analyzes markdown files for token efficiency. TRIGGERS: optimize markdown, reduce tokens, token count, token bloat, too many tokens, make concise, shrink file, file too large, optimize for AI, token efficiency, verbose markdown, reduce file size"
+---
+
+# Markdown Token Optimizer
+
+This skill analyzes markdown files and suggests optimizations to reduce token consumption while maintaining clarity.
```
Comment (praveenkuttappan) [inline .github/skills/markdown-token-optimizer/SKILL.md]: Is this to help writing the skills? If that's the case, can we move this also into a different PR where skills sensei and eval changes are included? Idea is to have one PR typespec to SDK workflow skills and another one for tools and test for the skills.

File: .github/skills/generate-sdk-locally/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/generate-sdk-locally/SKILL.md:1]: This skill also should be moved to azure-sdk-tools and make it shared across all repos.

File: .github/skills/generate-sdk-locally/SKILL.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
+name: generate-sdk-locally
+license: MIT
+metadata:
+  version: "1.0.0"
+description: >-
+  **UTILITY SKILL**
+  Generate, build, and test Azure SDKs locally from TypeSpec.
+  USE FOR: "generate SDK locally", "build SDK", "run SDK tests", "update changelog".
```
Comment (praveenkuttappan) [inline .github/skills/generate-sdk-locally/SKILL.md]:
```suggestion
  USE FOR: "generate SDK locally", "build SDK", "run SDK tests", "update changelog", "update metadata", "update version", "validate package", "check package"
```

File: .github/skills/check-package-validation/SKILL.md
Comment (praveenkuttappan) [inline .github/skills/check-package-validation/SKILL.md:1]: Should this be a skill? It is just one MCP tool call and it depends on sdk generation locally.

File: .github/skills/report-test-bug/SKILL.md (hunk 10 above)
Comment (ronniegeraghty) [reply .github/skills/report-test-bug/SKILL.md]: Removed report-test-bug from this PR. Will fix the Azure SDK tools repo name when it moves there.

File: .github/skills/markdown-token-optimizer/SKILL.md (hunk 11 above)
Comment (ronniegeraghty) [reply .github/skills/markdown-token-optimizer/SKILL.md]: Removed markdown-token-optimizer from this PR. It will come in a separate skill-development-tools PR along with sensei and skill-authoring.

File: .github/skills/package-release-readiness/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/package-release-readiness/SKILL.md:1]: Removed this skill. Its functionality is part of the sdk-release skill.

File: .github/skills/typespec-to-sdk-workflow/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/typespec-to-sdk-workflow/SKILL.md:1]: Removed this skill. The high-level workflow will go into AGENTS.md instead.

File: .github/skills/sensei/SKILL.md (hunk 4 above)
Comment (ronniegeraghty) [reply .github/skills/sensei/SKILL.md]: Removed the sensei skill from this PR. It will come in a separate skill-development-tools PR with these fixes applied.

File: .github/skills/validate-codeowners/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/validate-codeowners/SKILL.md:1]: Removed this skill from the PR. Will wait for Daniel's codeowners work.

File: .github/skills/verify-setup/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/verify-setup/SKILL.md:1]: Agreed — removed this skill. It can be referenced directly in AGENTS.md or folded into the generate-sdk-locally skill.

File: .github/skills/validate-service-label/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/validate-service-label/SKILL.md:1]: Removed this skill from the PR.

File: .github/skills/report-test-bug/SKILL.md (hunk 8 above)
Comment (ronniegeraghty) [reply .github/skills/report-test-bug/SKILL.md]: Removed report-test-bug from this PR. Will fix the skill name input issue when it moves to azure-sdk-tools.

File: .github/skills/report-test-bug/SKILL.md (hunk 9 above)
Comment (ronniegeraghty) [reply .github/skills/report-test-bug/SKILL.md]: Removed report-test-bug from this PR. Will address Azure-specific terminology when it moves to azure-sdk-tools.

File: .github/skills/sdk-release/SKILL.md (hunk 6 above)
Comment (ronniegeraghty) [reply .github/skills/sdk-release/SKILL.md]: Applied this change — updated step 4 to show the release pipeline link to the user and inform them they must approve the release stage after triggering.

File: .github/skills/sensei/SKILL.md (hunk 5 above)
Comment (ronniegeraghty) [reply .github/skills/sensei/SKILL.md]: Removed the sensei skill from this PR. Will address the Azure SDK tools references when it comes back in a future PR.

File: .github/skills/report-test-bug/SKILL.md (hunk 7 above)
Comment (ronniegeraghty) [reply .github/skills/report-test-bug/SKILL.md]: Removed report-test-bug from this PR. Will fix the repo name and other issues when it moves to the azure-sdk-tools repo.

File: .github/skills/typespec-customization/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/typespec-customization/SKILL.md:1]: Removed this skill from the PR. Sameeksha can add eval/frontmatter to her TypeSpec customization PR.

File: .vscode/mcp.json (hunk 2 above)
Comment (ronniegeraghty) [reply .vscode/mcp.json]: Reverted mcp.json to match main.

File: .github/skills/generate-sdk-locally/SKILL.md (hunk 12 above)
Comment (ronniegeraghty) [reply .github/skills/generate-sdk-locally/SKILL.md]: Applied this USE FOR update to the generate-sdk-locally skill. The skill has been removed from this PR and will move to azure-sdk-tools, but this fix is included.

File: .github/skills/typespec-authoring/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/typespec-authoring/SKILL.md:1]: Removed this skill from the PR.

File: cspell.yaml (hunk 3 above)
Comment (ronniegeraghty) [reply cspell.yaml:68]: Removed the skills-eval.yml workflow and .waza.yaml from this PR. The cspell overrides have also been reverted since they were only needed for the removed skills. We'll split the eval workflow into a separate PR.

File: .github/skills/check-package-validation/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/check-package-validation/SKILL.md:1]: Agreed — removed this skill. Its functionality is folded into the generate-sdk-locally skill.

File: .github/skills/report-test-bug/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/report-test-bug/SKILL.md:1]: Removed report-test-bug from this PR for now. We'll move it to azure-sdk-tools with a `shared-` prefix. Question: where should shared skills live in the azure-sdk-tools repo? Is there a common directory decided on yet, e.g. `eng/common/skills/` or `tools/azsdk-cli/skills/`?

File: .github/skills/generate-sdk-locally/SKILL.md
Comment (ronniegeraghty) [reply .github/skills/generate-sdk-locally/SKILL.md:1]: Removed generate-sdk-locally from this PR. We'll move it to azure-sdk-tools with a `shared-` prefix. Applied the USE FOR update from the other comment. Question: where should shared skills live in the azure-sdk-tools repo? Has a common directory been decided on yet (e.g. `eng/common/skills/` or `tools/azsdk-cli/skills/`)?

**Comment by ronniegeraghty** [comment]
Closing in favor of adding skills to Azure/azure-sdk-tools per https://github.com/Azure/azure-sdk-tools/issues/14261. Skills will be in eng/common/ with shared- prefix.
